File: frontend/views/misc/universal-footer.ts
/*
 * Copyright © 2021 NeuroByte Tech. All rights reserved.
 *
 * NeuroByte Tech is the Developer Company of Rohan Mathew.
 *
 * Project: latinvocab
 * File Name: universal-footer.ts
 * Last Modified: 08/04/2021, 21:32
 */

import {css, customElement, html, LitElement} from 'lit-element';
import '@vaadin/vaadin-ordered-layout/src/vaadin-horizontal-layout.js';

@customElement('universal-footer')
export class UniversalFooter extends LitElement {
  static get styles() {
    return css`
      :host {
          display: block;
          height: 100%;
      }
      `;
  }

  render() {
    return html`
      <vaadin-horizontal-layout class="footer"
                                style="width: 100%; flex-basis: var(--lumo-size-l); flex-shrink: 0; background-color: var(--lumo-contrast-10pct); justify-content: center;">
        <h6 id="cpyr">Copyright</h6>
      </vaadin-horizontal-layout>
    `;
  }

  // Remove this method to render the contents of this view inside Shadow DOM
  createRenderRoot() {
    return this;
  }
}

File: src/main/java/tech/neurobyte/dev/views/misc/UniversalFooter.java
/*
 * Copyright © 2021 NeuroByte Tech. All rights reserved.
 *
 * NeuroByte Tech is the Developer Company of Rohan Mathew.
 *
 * Project: latinvocab
 * File Name: UniversalFooter.java
 * Last Modified: 25/04/2021, 22:06
 */

package tech.neurobyte.dev.views.misc;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.html.H6;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.router.Route;

import java.time.LocalDate;

@Route("footer")
@Tag("universal-footer")
@JsModule("./views/misc/universal-footer.ts")
public class UniversalFooter extends LitTemplate {

    @Id("cpyr")
    private H6 cpyr;

    public UniversalFooter() {
        cpyr.setText(String.format("Copyright © %d Rohan Mathew. All rights reserved.", LocalDate.now().getYear()));
    }

}
